src/utils/getDotnetInfo.ts

```diff
@@ -13,15 +15,28 @@ let _dotnetInfo: DotnetInfo;
 // is new enough for us.
 // Returns: a promise that returns a DotnetInfo class
 // Throws: An DotNetCliError() from the return promise if either dotnet does not exist or is too old.
-export async function getDotnetInfo(): Promise<DotnetInfo> {
+export async function getDotnetInfo(dotNetCliPaths: string[]): Promise<DotnetInfo> {
     if (_dotnetInfo !== undefined) {
         return _dotnetInfo;
     }
 
+    let dotnetExeName = CoreClrDebugUtil.getPlatformExeExtension();
+    let dotnetExecutablePath = undefined;
+
+    for (const dotnetPath of dotNetCliPaths) {
+        let dotnetFullPath = join(dotnetPath, dotnetExeName);
+        if (CoreClrDebugUtil.existsSync(dotnetFullPath)) {
+            dotnetExecutablePath = dotnetFullPath;
+            break;
+        }
+    }
+
     let dotnetInfo = new DotnetInfo();
 
     try {
-        let data = await execChildProcess('dotnet --info', process.cwd());
+        let data = await execChildProcess(`${dotnetExecutablePath || 'dotnet'} --info`, process.cwd());
+
+        dotnetInfo.CliPath = dotnetExecutablePath; 
```

Comment:
Just wanted to ensure that the intention is this will be `undefined` when falling back to running `dotnet` from PATH.

Reply:
Yes, this is intentional.